src/anndata/_core/sparse_dataset.py

```diff
@@ -313,6 +316,13 @@ def mean_slice_length(slices):
 def get_memory_class(
     format: Literal["csr", "csc"], *, use_sparray_in_io: bool = False
 ) -> type[_cs_matrix]:
+    if not CAN_USE_SPARSE_ARRAY and use_sparray_in_io:
+        msg = (
+            "scipy.sparse.cs{r,c}array is not available in current scipy version."
+            "falling back to scipy.sparse.spmatrix for reading"
+        )
+        warnings.warn(msg, UserWarning, stacklevel=2)
+        use_sparray_in_io = False
```

Comment:
why not just put that into the setting.

if someone tries to set the setting to `True` when sparse arrays aren’t available, just throw an error instead.